test/token/otcEscrow.spec.ts

```diff
@@ -0,0 +1,297 @@
+        subjectVestingCliff,
+        subjectVestingEnd,
+        subjectUSDCAmount,
+        subjectIndexAmount,
+        usdc.address,
+        index.address
+      );
+      
+      await index.connect(indexGov.wallet).transfer(subjectOtcEscrow.address, subjectIndexAmount);
+      await usdc.connect(investor.wallet).approve(subjectOtcEscrow.address, subjectUSDCAmount);
+    });
+
+    async function subject(): Promise<ContractTransaction> {
+      return await subjectOtcEscrow.connect(indexGov.wallet).swap();
+    }
+
+    it("should send the investor's usdc to indexGov", async () => {
+      const initInvestorUsdc = await usdc.balanceOf(investor.address);
+      const initEscrowIndex = await index.balanceOf(subjectOtcEscrow.address);
+
+      await subject();
+
+      const finalInvestorUsdc = await usdc.balanceOf(investor.address);
+      const finalEscrowIndex = await index.balanceOf(subjectOtcEscrow.address);
+
+      expect(initEscrowIndex.sub(finalEscrowIndex)).to.eq(subjectIndexAmount);
+      expect(initInvestorUsdc.sub(finalInvestorUsdc)).to.eq(subjectUSDCAmount);
+    });
+
+    it("it should emit a VestingDeployed event", async () => {
+      await expect(subject()).to.emit(subjectOtcEscrow, "VestingDeployed");
+    });
+
+    it("it should transfer index to the vesting contract", async () => {
+      await subject();
+
+      const vestingDeployed = (await subjectOtcEscrow.queryFilter(subjectOtcEscrow.filters.VestingDeployed(null)))[0];
+      const vestingAddress = vestingDeployed.args?.vesting;
+
+      expect(await index.balanceOf(vestingAddress)).to.eq(subjectIndexAmount);
+    });
+
+    it("it should set the state variables of the vesting contract correctly", async () => {
+      await subject();
+
+      const vestingDeployed = (await subjectOtcEscrow.queryFilter(subjectOtcEscrow.filters.VestingDeployed(null)))[0];
+      const vestingAddress = vestingDeployed.args?.vesting;
+
+      const vesting = Vesting__factory.connect(vestingAddress, getProvider());
+
+      expect(await vesting.index()).to.eq(index.address);
+      expect(await vesting.recipient()).to.eq(investor.address);
+      expect(await vesting.vestingAmount()).to.eq(subjectIndexAmount);
+      expect(await vesting.vestingBegin()).to.eq(subjectVestingStart);
+      expect(await vesting.vestingCliff()).to.eq(subjectVestingCliff);
+      expect(await vesting.vestingEnd()).to.eq(subjectVestingEnd);
+    });
+
+    context("when the caller is unauthorized", async () => {
+
+      let subjectCaller: Account;
+
+      beforeEach(async () => {
+        subjectCaller = await getRandomAccount();
+      });
+
+      async function subject(): Promise<ContractTransaction> {
+        return await subjectOtcEscrow.connect(subjectCaller.wallet).swap();
+      }
+
+      it("should revert", async () => {
+        await expect(subject()).to.be.reverted;
+      });
+    });
+
+    context("when an inadequate amount of INDEX is sent to the escrow", async () => {
```

Comment:
Instead of a silent revert maybe we can add an explicit check in the function to make sure that there's enough INDEX balance